### ISSUE TITLE
`lowercase-name` rule `ignoreTopLevelDescribe` option only applies to first `describe` in file

### ISSUE THREAD

**Issue by overlookmotel** [jest-community/eslint-plugin-jest#613]
The `ignoreTopLevelDescribe` option for `lowercase-name` rule, added in #611, seems to only apply to the first `describe` block in a file.

e.g. `'Booleans'` gets flagged as a rule violation here:

```js
describe('Strings', () => {
  it('are strings', () => {
    expect('abc').toBe('abc');
  });
});

describe('Booleans', () => {
  it('are booleans', () => {
    expect(true).toBe(true);
  });
});
```

**Comment by G-Rath** [comment]
Whoops that's my bad - I forgot to decrease the describe counter when going back up the nest 😬
